### PR TITLE
test: Ensure only one thread advances time, and enable simulated-time for hds_integration_test.

### PR DESCRIPTION
**Pull request by jmarantz**
Description: It is hard to reason about tests that push time forward from more than one thread. This was happening in contention-tests. This adds some asserts to ensure that we don't do this in future tests, and fixes those tests that were doing it.
Risk Level: low
Testing: //test/..., and hds_integration_test witih tsan and --runs_per_test=1000
Docs Changes: n/a
Release Notes: n/a
Fixes: #6239

### CODE REVIEW

**Comment by jmarantz** [comment]
It's possible this may fix #5183 as well.

File: test/test_common/utility_core.cc [hunk 1]
```diff
@@ -0,0 +1,35 @@
+#include "common/common/thread_impl.h"
```
Comment (lizan) [inline test/test_common/utility_core.cc]: Are these two extracted for avoid circular dependency? The "utility_core" vs "utility" split sounds confusing. Any idea for a better naming? "filesystem.h" and "thread.h"?

Comment (jmarantz) [reply test/test_common/utility_core.cc]: Exactly - split for circular dependency. Yes I can split them.

Comment (lizan) [reply test/test_common/utility_core.cc]: OK let's do that, other than this the PR LGTM.

Comment (jmarantz) [reply test/test_common/utility_core.cc]: done.

**Pull request merged**